core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider.go

```
@@ -92,36 +104,33 @@ func (provider *GitPackageContentProvider) GetOnDiskAbsoluteFilePath(fileInsideP
 	if interpretationError != nil {
 		return "", interpretationError
 	}
-	return pathToFile, nil
-}
-
-func (provider *GitPackageContentProvider) GetModuleContents(fileInsideModuleUrl string) (string, *startosis_errors.InterpretationError) {
-	pathToFile, interpretationError := provider.GetOnDiskAbsoluteFilePath(fileInsideModuleUrl)
-	if interpretationError != nil {
-		return "", interpretationError
-	}
 
-	maybeKurtosisYamlPath, err := checkIfFileIsInAValidPackage(pathToFile, provider.packagesDir)
+	// check whether kurtosis yaml exists in th path
+	maybeKurtosisYamlPath, err := getOptionalKurtosisYmlPathForFileUrl(pathToFile, provider.packagesDir)
 	if err != nil {
-		return "", startosis_errors.WrapWithInterpretationError(err, "Error occurred while verifying whether '%v' belongs to a Kurtosis package.", fileInsideModuleUrl)
+		return "", startosis_errors.WrapWithInterpretationError(err, "Error occurred while verifying whether '%v' belongs to a Kurtosis package.", fileInsidePackageUrl)
 	}
 
 	if maybeKurtosisYamlPath == filePathToKurtosisYamlNotFound {
-		return "", startosis_errors.NewInterpretationError("%v is not found in the path of '%v'; files can only be imported or read from Kurtosis packages. For more information, go to: %v", startosis_constants.KurtosisYamlName, fileInsideModuleUrl, howImportWorksLink)
+		return "", startosis_errors.NewInterpretationError("%v is not found in the path of '%v'; files can only be accessed from Kurtosis packages. For more information, go to: %v", startosis_constants.KurtosisYamlName, fileInsidePackageUrl, howImportWorksLink)
```

curious can we simply throw an error from `getOptionalKurtosisYmlPathForFileUrl` if we can't access it / doesn't exist? is that function useful if it doesn't provide yml

```
getKurtosisYmlPathForFile(filePath, packagesDir) -> ymlPath, err {
/// 
if errorfindingyml {
    return errorfindingyml
}

if path == empty {
   return ymlNotFoundErr
}

return ymlPath, nil
}
```

Thoughts?

---

i leave it for now; only called once.. i did the error outside cause i had access `fileInsidePackageUrl` for nice error message.. i could always pass it down the method `getKurtosisYmlPathForFile` but seemed unnecessary.. also did not wanted to wrap both error messages ( not found and unexpected err) into one as well.